README.md

```diff
@@ -279,6 +280,107 @@ module.exports = {
 }
 ```
 
+### Asynchronous code
+
+Asynchronous code can be achieved by using `async` methods:
+
+```js
+// index.js
+
+module.exports = {
+  onPreBuild: async ({ utils }) => {
+    try {
+      await doSomethingAsync()
+    } catch (error) {
+      utils.build.failBuild('Failure message', { error })
+    }
+  },
+}
+```
+
+Any thrown `Error` or rejected `Promise` that is not handled by [`utils.build`](#error-reporting) will be reported as
+bugs.
```

Comment:
What does it mean to report something as a bug in this context? You will file issues on their repos? 😉

Reply:
Oh yes, that's not clear. 
By "reported", I meant printed in the build lots. And by "as bugs", I meant the logs will say this is a bug from the plugin, as opposed to a user error.
I tried to clarify in 1cc01656537f60bf1773f05df78cb3716b90d553, please let me know what you think.